Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,437 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Plugin.Calendars.Abstractions;
+using Tizen.Pims.Calendar;
+using Tizen.Pims.Calendar.CalendarViews;
+
+namespace Plugin.Calendars
+{
+	/// <summary>
+	/// Implementation for Calendars
+	/// </summary>
+	public class CalendarsImplementation : ICalendars
+	{
+		/// <summary>
+		/// Gets a list of all calendars on the device.
+		/// </summary>
+		/// <returns>Calendars</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<Calendar>> GetCalendarsAsync()
+		{
+			return Task.Run<IList<Calendar>>(() =>
+			{
+				var calendars = new List<Calendar>();
+
+				CalendarRecord Crecord = new CalendarRecord(Book.Uri);
+				Crecord.Set<string>(Book.Name, "org.tizen.calendar");
+
+				calendars.Add(
+					new Calendar
+					{
+						Name = Crecord.Get<string>(Book.Name),
+						ExternalID = Crecord.Get<int>(Book.Id).ToString(),
+						CanEditEvents = true,
+						CanEditCalendar = false,
+						AccountName = Crecord.Get<int>(Book.AccountId).ToString(),
+						Color = Crecord.Get<string>(Book.Color),
+					}
+				);
+
+				Crecord?.Dispose();
+				Crecord = null;
+
+				return calendars;
+			});			
+		}
+
+		/// <summary>
+		/// Gets a single calendar by platform-specific ID.
+		/// </summary>
+		/// <param name="externalId">Platform-specific calendar identifier</param>
+		/// <returns>The corresponding calendar, or null if not found</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<Calendar> GetCalendarByIdAsync(string externalId)
+		{
+			CalendarRecord Crecord = new CalendarRecord(Book.Uri);
+			Crecord.Set<string>(Book.Name, "org.tizen.calendar");
+
+			Calendar calendar = new Calendar
+			{
+				Name = Crecord.Get<string>(Book.Name),
+				ExternalID = Crecord.Get<int>(Book.Id).ToString(),
+				CanEditEvents = true,
+				CanEditCalendar = false,
+				AccountName = Crecord.Get<int>(Book.AccountId).ToString(),
+				Color = Crecord.Get<string>(Book.Color),
+			};
+
+			Crecord?.Dispose();
+			Crecord = null;
+
+			return Task.Run(() => calendar);
```

Comment:
I think you just want to be using `Task.FromResult` instead of `Task.Run` if you don't actually have logic that should be run in the background.